constraints/src/main/java/gov/nasa/jpl/aerie/constraints/time/Spans.java

```diff
@@ -174,7 +174,7 @@ public boolean isCollectionSubsetOf(final Spans otherSpans){
 
   public Spans intersectWith(final Windows windows){
     final var ret = new Spans();
-    this.intervals.forEach(x -> windows.iterator().forEachRemaining(y -> ret.add(Interval.intersect(y.interval(), x.interval()), x.value())));
+    this.intervals.forEach(x -> windows.iterateEqualTo(true).iterator().forEachRemaining(y -> ret.add(Interval.intersect(x.interval(), y), x.value())));
```

Comment:
I'd like to see a quick test of this method that would fail with the previous implementation but succeed with this fix, both because it's weird that all of our tests used to pass and to ensure we don't add back the same bug in the future.

Reply:
done